src/DependencyManagement/DependencyManager.cs

```diff
@@ -0,0 +1,372 @@
+            {
+                // The requested major version is greater than the latest major supported version.
+                var errorMsg = string.Format(PowerShellWorkerStrings.InvalidModuleMajorVersion, moduleName, majorVersion);
+                var argException = new ArgumentException(errorMsg);
+                throw argException;
+            }
+        }
+
+        /// <summary>
+        /// Parses the given string version and extracts the major version.
+        /// Please note that the only version we currently support is of the form '1.*'.
+        /// </summary>
+        private string GetMajorVersion(string version)
+        {
+            if (string.IsNullOrEmpty(version))
+            {
+                var errorMessage = String.Format(PowerShellWorkerStrings.DependencyPropertyIsNullOrEmpty, "version");
+                throw new ArgumentException(errorMessage);
+            }
+
+            // Validate that version is in the correct format: 'MajorVersion.*'
+            if (!IsValidVersionFormat(version))
+            {
+                var errorMessage = String.Format(PowerShellWorkerStrings.InvalidVersionFormat, "MajorVersion.*");
+                throw new ArgumentException(errorMessage);
+            }
+
+            // Return the major version.
+            return version.Split(".")[0];
+        }
+
+        /// <summary>
+        /// Parses the given PowerShell (psd1) data file.
+        /// Returns a Hashtable representing the key value pairs.
+        /// </summary>
+        private Hashtable ParsePowerShellDataFile(string functionAppRootPath, string fileName)
+        {
+            // Path to Requirements.psd1 file.
+            var requirementsFilePath = Path.Join(functionAppRootPath, fileName);
+
+            // Validate the file path.
+            if (!File.Exists(requirementsFilePath))
+            {
+                var errorMessage = String.Format(PowerShellWorkerStrings.FileNotFound, fileName, functionAppRootPath);
+                throw new ArgumentException(errorMessage);
+            }
+
+            // Try to parse the Requirements.psd1 file.
+            var ast = Parser.ParseFile(requirementsFilePath, out _, out ParseError[] errors);
+
+            if (errors?.Length > 0)
+            {
+                var stringBuilder = new StringBuilder();
+                foreach (var error in errors)
+                {
+                    stringBuilder.AppendLine(error.Message);
+                }
+
+                string errorMsg = stringBuilder.ToString();
+                throw new ArgumentException(string.Format(PowerShellWorkerStrings.FailToParseScript,
+                    RequirementsPsd1FileName, errorMsg));
+            }
+
+            var hashtableAst = ast.Find(x => x is HashtableAst, false);
+            var hashtable = default(Hashtable);
+
+            bool throwException = false;
+            try
+            {
+                hashtable = (Hashtable)hashtableAst?.SafeGetValue();
```

Comment:
I would assume `SafeGetValue` shouldn't throw - Is it not safe after all? :)

Reply:
You are right, this method does not throw. If any parsing issues occurred, these are caught by ```Parser.ParseFile``` I updated the code. Also, I am using the ```as``` keyword to cast the ```hashtableAst?.SafeGetValue()``` to a ```Hashtable``` and evaluate the result to null.